7_deploy_osh/roles/suse-dev-patcher/tasks/patch-review-repos.yml

```diff
@@ -0,0 +1,15 @@
+---
+
+- name: Apply changesets in flight for review site {{ review_site }}
+  shell: |
```

Comment:
I'd use a `set` to fail on any error (including pipe), else it can be silently passing through until the last line.

This can be re-implemented in ansible with one task with the uri module to fetch data from the API, and another task to fetch/apply the cherry-pick. As this was also an include with a single task, you could simplify the readability of the tasks by looping direcly on the include directive. Simple and very readable.

But ofc, I'd prefer if that was a simple ansible module (or action plugin, for laziness) instead :)

Reply:
I will add set -e in beginning of shell commands.

For loop aspect, here we are looping over all patches within a review site ( e.g. review.openstack.org). That review site may have multiple patches per repo. So this is looping at repo and then at patches within that repo.

Outer loop (at include task level) is per  review site.
Given above info, this current logic works now. I am not too sure if I am understand the above suggestion. Can it be clarified little more or may be we can improve in subsequent patch if needed. As dev-patcher functonality is needed for other dependent tasks so better merge this patch if there are no significant issue.

Reply:
Indeed, it can be improved later.